code/game/machinery/vending/cm_vending.dm

```diff
@@ -440,6 +440,43 @@ GLOBAL_LIST_EMPTY(vending_products)
 	user.set_interaction(src)
 	tgui_interact(user)
 
+/// Handles redeeming coin tokens.
+/obj/structure/machinery/cm_vending/attackby(obj/item/attacking_item, mob/user)
+	if(!istype(attacking_item, /obj/item/coin/marine))
+		..()
+	if(!can_access_to_vend(user, ignore_hack = TRUE))
+		return FALSE
+	redeem_token(attacking_item, user)
+
+/obj/structure/machinery/cm_vending/proc/get_token_type(obj/item/coin/marine/token)
```

Comment:
This could just be a var on the token subtype

Reply:
DOH